### ISSUE TITLE
[BUG] Detrender in TransformedTargetForecaster does not work

### ISSUE THREAD

**Issue by shagn** [alan-turing-institute/sktime#2739]
**Describe the bug**
When using a Detrender in a TransformedTargetForecaster and trying to create predictions, the inverse transformation seems to fail and therefore no predictions can be made. 
Detrender and Forecaster work fine when not concatenated in a TransformedTargetForecaster. 

**To Reproduce**
```python
from sktime.forecasting.trend import PolynomialTrendForecaster
from sktime.transformations.series.detrend import Detrender
from sktime.forecasting.compose import TransformedTargetForecaster
from sktime.forecasting.naive import NaiveForecaster
from sktime.datasets import load_airline

y = load_airline()

trans_fc = TransformedTargetForecaster(
    [
        ("detrender", Detrender(forecaster=PolynomialTrendForecaster(degree=1))),
        ("forecaster", NaiveForecaster(strategy="last")),
    ]
)
trans_fc.fit(y)
trans_fc.predict(1)
```
![image](https://user-images.githubusercontent.com/16029092/171620553-68a421e8-a9b5-4dac-a99c-ff7a406875de.png)

**Expected behavior**
The inverse transformation of a Detrender should work in a TransformedTargetForecaster. 

**Versions**
<details>
System:
    python: 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0]
executable: /anaconda/envs/hog-forecast/bin/python
   machine: Linux-5.4.0-1056-azure-x86_64-with-glibc2.17

Python dependencies:
          pip: 21.2.4
   setuptools: 61.2.0
      sklearn: 1.0.2
       sktime: 0.11.4
  statsmodels: 0.13.2
        numpy: 1.22.3
        scipy: 1.7.3
       pandas: 1.4.2
   matplotlib: 3.5.1
       joblib: 1.1.0
        numba: 0.53.1
     pmdarima: None
      tsfresh: None

</details>

**Comment by fkiraly** [comment]
confirmed with current main, python 3.8 windows.

Problem looks like trouble with column renaming in the `Detrender`, combined with `sklearn`'s new input check on equal column names.

**Comment by fkiraly** [comment]
also fails with other forecasters instead of `PolynomialTrendForecaster`, so not a problem of that

**Comment by fkiraly** [comment]
I think I understand the root cause.

The problem materializes under the following conditions:
* a `pd.Series` `y` is converted to `pd.DataFrame`, then back to `pd.Series`, then back to `pd.DataFrame`, by `convert` or `convert_to`
* the initial series `y` has a `name` attribute set
* there can be "pairing" through `converter_store`, but it is between the first two, not the last two conversions
* there can be transformations in-between the conversions, as long as they do not change column names or type

Then, what happens:
* after the first conversion, we have a `pd.DataFrame` with column name `y.name`
* after the last conversion, we have a `pd.DataFrame` with column name `0`

I.e., the name gets lost, and the column names are inconsistent between the two occurrences.

If, in addition, the conversions occur in a context where naming consistency is assumed (like in the pipeline above), an exception is raised.

**Comment by fkiraly** [comment]
I see two potential fixes, via the conversions:

* we do not allow `pd.Series` to have `name` at all. As soon as we encounter `pd.Series`, we remove the `name` attribute. Attempted here: https://github.com/alan-turing-institute/sktime/pull/2788
* we ensure we keep the `name` attribute consistent throughout conversions, in this case we need to change the conversion from `pd.DataFrame` to `pd.Series` which eats the `name` away after conversion. Attempted here: https://github.com/alan-turing-institute/sktime/pull/2789

**Comment by fkiraly** [comment]
Both seem to work, I'll add a test

**Comment by fkiraly** [comment]
Hm, it looks like "keeping the `name` attribute" #2788 fixes *this* bug, but creates other bugs...

**Comment by fkiraly** [comment]
Fix no.3: doesn't break anything, but also doesn't fix this.
https://github.com/alan-turing-institute/sktime/pull/2791

**Comment by fkiraly** [comment]
This fix to `NaiveForecaster` apparently fixes the problem:
https://github.com/alan-turing-institute/sktime/pull/2793

**Comment by fkiraly** [comment]
This is fixed now via #2793 and should work in 0.12.0.
It has also highlighted inconsistent handling of `pd.Series.name` which we need to think about carefully.
